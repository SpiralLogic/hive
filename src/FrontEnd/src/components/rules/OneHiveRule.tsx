import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';

export default {
  title: 'One Hive Rule',
  description: ['There must only be 1 hive', 'Even in transit a hive must not be split'],
  Rule: () => (
    <Hextille class="rules">
      <Row>
        <RuleCell creature="beetle" />
        <RuleCell creature="beetle" />
      </Row>
      <Row>
        <RuleCell creature="spider" />
        <RuleCell />
        <RuleCell result="incorrect" />
        <RuleCell />
      </Row>
      <Row>
        <RuleCell creature="ant" />
        <RuleCell creature="queen" incorrectArrows={['topRight']} />
        <RuleCell creature="grasshopper" />
      </Row>
    </Hextille>
  ),
};
