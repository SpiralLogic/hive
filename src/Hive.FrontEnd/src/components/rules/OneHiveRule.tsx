import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';
import { Rule } from './rule';

const OneHiveRule: Rule = () => (
  <Hextille>
    <Row>
      <RuleCell creature="beetle" />
      <RuleCell creature="beetle" />
    </Row>
    <Row>
      <RuleCell creature="spider" />
      <RuleCell />
      <RuleCell result="incorrect" />
      <RuleCell />
    </Row>
    <Row>
      <RuleCell creature="ant" />
      <RuleCell selected creature="queen" incorrectArrows={['top-right']} />
      <RuleCell creature="grasshopper" />
    </Row>
  </Hextille>
);

OneHiveRule.title = 'One Hive Rule';
OneHiveRule.description = ['There must only be 1 hive', 'Even in transit a hive must not be split'];
OneHiveRule.displayName = 'OneHiveRule';

export default OneHiveRule;
