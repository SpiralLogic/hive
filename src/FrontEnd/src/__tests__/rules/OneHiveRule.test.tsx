import { h } from 'preact';
import OneHiveRule from '../../components/rules/OneHiveRule';
import { renderElement } from '../helpers';

describe('One Hive Rule snapshot tests', () => {
  test('snapshot', () => {
    expect(renderElement(<OneHiveRule />)).toMatchSnapshot();
  });
});
