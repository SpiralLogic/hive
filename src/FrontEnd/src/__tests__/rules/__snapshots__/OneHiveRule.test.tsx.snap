// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One Hive Rule snapshot tests snapshot 1`] = `
<div
  class="left"
  role="grid"
>
  <div
    role="row"
  >
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile beetle"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#beetle"
          />
        </svg>
        
      </div>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile beetle"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#beetle"
          />
        </svg>
        
      </div>
    </div>
  </div>
  <div
    role="row"
  >
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile spider"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#spider"
          />
        </svg>
        
      </div>
    </div>
    <div
      class="blank"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        
      </div>
    </div>
    <div
      class="incorrect"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          <span>
            âœ˜
          </span>
        </span>
      </div>
    </div>
    <div
      class="blank"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        
      </div>
    </div>
  </div>
  <div
    role="row"
  >
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#ant"
          />
        </svg>
        
      </div>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile queen"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#queen"
          />
        </svg>
        
      </div>
      <svg
        class="arrow topRight incorrect"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile grasshopper"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#grasshopper"
          />
        </svg>
        
      </div>
    </div>
  </div>
</div>
`;
